### PR TITLE
pthread-emb pkgbuild

### PR DESCRIPTION
**Pull request by Sir-Fortesque**
Create a pkgbuild file for pthread-emb.

The original script (https://github.com/pspdev/psplibraries/blob/master/scripts/pthreads-emb.sh) pulled from the following repo https://github.com/take-cheeze/pthreads-emb 

I made a change to instead pull from the sourceforge project (https://sourceforge.net/projects/pthreads-emb/) and apply a patch that is included in the pull request. I did this as the `Makefile` for the project hardcodes the install location from `psp-config` binary. As I needed to apply a patch to allow a supllied install location, I lumped all the changes in the patch.

I also made the `Makefile` a little nicer with variables for lib and include directories as well as using the `install` to copy the created library files.

### CODE REVIEW

File: pthreads-emb/PSPBUILD [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+pkgrel=1
+pkdesc="A pthreads library for embedded operating systems."
+arch=("mips")
+url="http://pthreads-emb.sourceforge.net/"
+license=("(L)GPL2")
+depends=()
+optdepends=()
+makedepends=()
+source=(
+    "https://excellmedia.dl.sourceforge.net/project/${pkgname}/${pkgname}/${pkgver}/${pkgname}-${pkgver}.tar.gz"
```
Comment (carstene1ns) [inline pthreads-emb/PSPBUILD:13]: Please do not use a specific sf.net mirror directly. These links get outdated, better to use a redirection link:
`http://downloads.sourceforge.net/pthreads-emb/pthreads-emb-1.0.tar.gz`

**Comment by carstene1ns** [comment]
I would suggest we take the patch and original source and have this in pspdev/pthreads repository instead.

File: pthreads-emb/PSPBUILD (hunk 1 above)
Comment (Sir-Fortesque) [reply pthreads-emb/PSPBUILD:13]: Thanks for the heads up. I tend to stray away from sourceforge and so I'm not particular to the working of it.

**Comment by Sir-Fortesque** [comment]
> I would suggest we take the patch and original source and have this in pspdev/pthreads repository instead.

@carstene1ns I agree with you about keeping the source within pspdev. How would you suggest that we proceed with this? 

I see the best way to do this is for to me to make some changes to the makefile within my pthreads-emb fork, trasfer the repository to pspdev. Make a release available for download. Finally modify this package file to use a release tar instread.

**Pull request closed without merging**